### ISSUE TITLE
Feedback on draft-richer-transactional-authz-04

### ISSUE THREAD

**Issue by Govvy** [bspk/oauth.xyz-site#15]
Is this the correct place to raise questions/propose modification, I didnt see this draft being posted on the mailing list for review yet...

1. I note that you are using protocol, I assume this is as opposed to the looser framework terminology of the original OAuth specs?

1.1. Pedantic but RC introduced before being defined, I sometimes see 'User Agent' terminology in OAuth 2.0 - is there any similar term to be defined?

1.2. Does any differentiation between user initiated sequences and system to system sequences need to be defined?

2. I think there is already an issue logged for this but perhaps 'display' is an ambiguous term

2.2. Do you intend to address the use case covered by the token exchange spec? I assume there is a situation in a microservices scenario where the token minted needs to be exchanged for a downstream set of services where the least privilege set of authorization information resources needs to be defined for each

2.4. Does DIDComm need to be defined somewhere? I assume its related to https://github.com/hyperledger/aries-rfcs/tree/master/concepts/0005-didcomm

3.3. Why this concatenation and separator choice, is there an argument for some alternative?

8.0. Will the access token be a similar model that would require an extension such as OIDC to issue an authentication token? If so, why couldnt both AuthZ and AuthN be defined in a single 'protocol'

**Comment by jricher** [comment]
This spec doesn't have a home working group yet, so either here, the oauth.xyz-spec tracker, or the TXAuth mailing list is fine. 

1.0) I think this is more of a protocol with branches instead of a collection of loosely related protocols in OAuth2. It's a different approach with similar modularity in the end.

1.1) I tried introducing "RC" as "something that calls the transaction endpoint" which is often the OAuth-style client or the RS depending on circumstances. I'm not sure that attempt has succeeded yet.

1.2) I don't see any difference that needs to be defined because they all start the same way. Main difference is that the client and AS need to know that user interaction isn't needed to approve a transaction, and that should be known by the resources and keys, not from a parameter.

2.0) Yes it is, suggestions for a better term for "things that get displayed to the end user during an interactive session at the AS?"

2.2) They're already covered. You can already push in information about the user, about the resources, and about the client. There's not a slot for it yet but you could easily also push in an access token, assuming you don't have a transaction handle instead of the token itself. 

2.4) DIDComm does not have a central definition. This might move to an extension but I wanted to show an example of non-HTTP communication, at least in theory.

3.3) Happy to entertain alternatives. This was chosen because the values in play won't naturally have newlines in them that aren't encoded, and we could define them not to, very easily.

8.0) No, you don't extend the access token. Instead you extend the transaction response, which includes the access token. Aaron Parecki has a blog post about this: https://aaronparecki.com/2019/07/18/17/adding-identity-to-xyz

**Comment by Govvy** [comment]
How about 'client_info' or 'resource_client_info':

client_info    Information about the RC making the request, including
      display name, home page, logo, and other user-facing information.
      This section is RECOMMENDED.

**Comment by Govvy** [comment]
Any chance we can also change the ‘nonce’ terminology too since it has an
unfortunate secondary meaning

**Comment by jricher** [comment]
What secondary meaning does "nonce" have? Here's it's a "number used only once", which is what "nonce" means.

**Comment by Govvy** [comment]
From google define:

nonce2
/nɒns/
Learn to pronounce
<https://www.google.com/search?safe=strict&client=mobilesearchapp&rlz=1MDAPLA_en-GBCH816CH816&source=mobilesearchapp&biw=375&hl=en_GB&channel=iss&v=82.1.267240167&bih=768&sxsrf=ACYBGNSWoSs1KsxsOy3G_PLmE0IXxbejGA:1577351914367&q=how+to+pronounce+nonce&stick=H4sIAAAAAAAAAOMIfcRozi3w8sc9YSm9SWtOXmPU4OINKMrPK81LzkwsyczPExLkYglJLcoV4pRi52LNy89LTrViUWJKzeNZxCqWkV-uUJKvUADUkA_UkaoAlgcALbTVelUAAAA&pron_lang=en&pron_country=gb&sa=X&ved=2ahUKEwimq6rk_dLmAhXDolwKHdtUBKgQ3eEDMAB6BAgBEBI>
*noun*
INFORMAL•BRITISH
noun: *nonce*; plural noun: *nonces*

   1. a person convicted of a sexual offence, especially against a child.

**Comment by jricher** [comment]
There are regional slang usages for just about every possible combination of noises that a human can make, though. I'm not inclined to jump through hoops when the term is established within technical and crypto circles to mean something specific.
